### PR TITLE
github action workflow

### PR DESCRIPTION
**Pull request by struga0258**
Signed-off-by: James Struga <xhejmsstruga@gmail.com>

## Proposed changes

This PR addresses Issue: [*Link to Github issue within https://github.com/zowe/zlux/issues* if any]

This PR depends upon the following PRs:

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Change in a documentation
- [ ] Refactor the code 
- [ ] Chore, repository cleanup, updates the dependencies.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## PR Checklist
Please delete options that are not relevant.
- [ ] If the changes in this PR are meant for the next release / mainline, this PR targets the "staging" branch.
- [ ] My code follows the style guidelines of this project (see: [Contributing guideline](https://github.com/zowe/zlux/blob/master/CONTRIBUTING.md))
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] video or image is included if visual changes are made
- [ ] Relevant update to CHANGELOG.md
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works, or describe a test method below

### CODE REVIEW

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+name: Build and Test Workflow
+on:
+  push:
+    branches:
+      - "*"
```
Comment (adityadr-joy) [inline .github/workflows/build_test.yml:5]: Is it worth to trigger workflow for push on all branches? Can we keep only major branches like master, staging, release candidates, release etc.

File: .github/workflows/build_test.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        uses: actions/checkout@v2
+        with: 
+            submodules: true
+
+      - name: '[Prep 2] Setup jFrog CLI'
+        uses: jfrog/setup-jfrog-cli@v2
+        env:
+          JF_ARTIFACTORY_1: ${{ secrets.JF_ARTIFACTORY_TOKEN }}
+          
+      - name: 'conver manifest'
```
Comment (adityadr-joy) [inline .github/workflows/build_test.yml:40]: spelling check 'convert manifest'

File: .github/workflows/build_test.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          pax-ssh-username: ${{ secrets.SSH_MARIST_USERNAME }}
+          pax-ssh-password: ${{ secrets.SSH_MARIST_RACF_PASSWORD }}
+
+      - name: '[Publish] Publish'
+        uses: zowe-actions/shared-actions/publish@main
+        if: success()
+        with:
+          artifacts: |
+            .pax/zss.pax
+            .pax/zss.tar
```
Comment (adityadr-joy) [inline .github/workflows/build_test.yml:78]: If "[Packaging] Make pax" step is not building .tar build then this line can be removed.

**Comment by struga0258** [comment]
@adityadr-joy I have addressed all the comments you had left, please take a look.

**Comment by adityadr-joy** [comment]
Looks good to me. Thanks!

**Review by 1000TurquoisePogs — APPROVED** [review]
merging for test.

**Pull request merged**